packages/core/src/index.tsx

```diff
@@ -11,12 +11,16 @@ import App from 'App/app.jsx';
 import { checkAndSetEndpointFromUrl } from '@deriv/shared';
 import AppNotificationMessages from './App/Containers/app-notification-messages.jsx';
 import './Utils/Datadog'; // to enable datadog
+import { isServiceWorkerSupported } from '@deriv/utils';
 
 if (
-    !!window?.localStorage.getItem?.('debug_service_worker') || // To enable local service worker related development
-    (!window.location.hostname.startsWith('localhost') && !/binary\.sx/.test(window.location.hostname)) ||
-    window.location.hostname === 'deriv-app.binary.sx'
+    isServiceWorkerSupported() &&
+    (!!window?.localStorage.getItem?.('debug_service_worker') || // To enable local service worker related development
+        (!window.location.hostname.startsWith('localhost') && !/binary\.sx/.test(window.location.hostname)) ||
+        window.location.hostname === 'deriv-app.binary.sx')
 ) {
+    // eslint-disable-next-line no-console
+    console.log('sw registered');
```

Comment:
this is for debugging purposes to check and make sure the service worker is being registered or not . I'll remove it